### PR TITLE
update request modifier fn type to impl send and sync

### LINKED ISSUE #769: v5::EventLoop not Send when websocket is enabled
# Expected Behavior

When enabling the feature "websocket", the v5::EventLoop should still be Send as it is the case when "websocket" is not enabled.

# Current Behavior

The v5::EventLoop is not Send anymore, even though the documentation states that the trait Send is implemented.

# Failure Information (for bugs)

```
error: future cannot be sent between threads safely
--> src\main.rs:24:24
|
24  |       tokio::task::spawn(async move {
|  ________________________^
25  | |         loop {
26  | |             match event_loop.poll().await {
27  | |                 Ok(event) => {
...   |
39  | |         }
40  | |     })
| |_____^ future created by async block is not `Send`
|
= help: the trait `Sync` is not implemented for `(dyn Fn(http::request::Request<()>) -> Pin<Box<(dyn Future<Output = http::request::Request<()>> + 'static)>> + 'static)`
note: captured value is not `Send`
--> src\main.rs:26:19
|
26  |             match event_loop.poll().await {
|                   ^^^^^^^^^^ has type `rumqttc::v5::EventLoop` which is not `Send`
note: required by a bound in `tokio::spawn`
--> C:\..path..to..\index.crates.io-6f17d22bba15001f\tokio-1.35.0\src\task\spawn.rs:166:21
|
164 |     pub fn spawn<F>(future: F) -> JoinHandle<F::Output>
|            ----- required by a bound in this function
165 |     where
166 |         F: Future + Send + 'static,
|                     ^^^^ required by this bound in `spawn`
```

## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Operating System: Windows
* Toolchain version: 1.73.0
* not tested with non-v5 AsyncClient, only v5

Repository with example code: https://github.com/kaans/rumqttc_websocket_error

## Failure Logs

Please include any relevant log snippets or files here.

### PR DESCRIPTION
**Pull request by swanandx**
Fixes #769 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] Formatted with `cargo fmt`
- [x] Make an entry to `CHANGELOG.md` if it's relevant to the users of the library. If it's not relevant mention why.

### CODE REVIEW

File: rumqttc/src/v5/mod.rs [hunk 1]
```diff
@@ -50,8 +50,9 @@ pub enum Request {
 }
 
 #[cfg(feature = "websocket")]
-type RequestModifierFn =
-    dyn Fn(http::Request<()>) -> Pin<Box<dyn Future<Output = http::Request<()>>>>;
+type RequestModifierFn = (dyn Fn(http::Request<()>) -> Pin<Box<dyn Future<Output = http::Request<()>> + Send>>
```
Comment (de-sh) [inline rumqttc/src/v5/mod.rs:53]:
```suggestion
type RequestModifierFn = Arc<
    dyn Fn(http::Request<()>) -> Pin<Box<dyn Future<Output = http::Request<()>> + Send>>
        + Send
        + Sync,
>;
```

File: rumqttc/src/v5/mod.rs [hunk 2]
```diff
@@ -50,8 +50,9 @@ pub enum Request {
...
+     + Send
```
Comment (de-sh) [inline rumqttc/src/v5/mod.rs:54]:
```suggestion
```

File: rumqttc/src/v5/mod.rs [hunk 3]
```diff
@@ -50,8 +50,9 @@ pub enum Request {
...
+     + Sync);
```
Comment (de-sh) [inline rumqttc/src/v5/mod.rs:55]:
```suggestion
```

File: rumqttc/src/v5/mod.rs [hunk 4]
```diff
@@ -100,7 +101,7 @@ pub struct MqttOptions {
     /// The server may set its own maximum inflight limit, the smaller of the two will be used.
     outgoing_inflight_upper_limit: Option<u16>,
     #[cfg(feature = "websocket")]
-    request_modifier: Option<Arc<Box<RequestModifierFn>>>,
+    request_modifier: Option<Arc<RequestModifierFn>>,
```
Comment (de-sh) [inline rumqttc/src/v5/mod.rs:104]:
```suggestion
    request_modifier: Option<RequestModifierFn>,
```

File: rumqttc/src/v5/mod.rs [hunk 5]
```diff
@@ -201,21 +202,22 @@ impl MqttOptions {
...
                 async move { request_modifier(request).into_future().await }
             })
-        })));
+        }));
         self
     }
 
     #[cfg(feature = "websocket")]
-    pub fn request_modifier(&self) -> Option<Arc<Box<RequestModifierFn>>> {
+    pub fn request_modifier(&self) -> Option<Arc<RequestModifierFn>> {
```
Comment (de-sh) [inline rumqttc/src/v5/mod.rs:220]:
```suggestion
    pub fn request_modifier(&self) -> Option<RequestModifierFn> {
```

**Review by de-sh — APPROVED** [review]

File: rumqttc/src/v5/mod.rs [hunk 6]
```diff
@@ -201,21 +202,22 @@ impl MqttOptions {
     #[cfg(feature = "websocket")]
     pub fn set_request_modifier<F, O>(&mut self, request_modifier: F) -> &mut Self
     where
-        F: Fn(http::Request<()>) -> O + 'static,
+        F: Fn(http::Request<()>) -> O + Sync + Send + 'static,
```
Comment (de-sh) [inline rumqttc/src/v5/mod.rs:205]:
```suggestion
    #[cfg(feature = "websocket")]
    pub fn set_request_modifier<F, O>(&mut self, request_modifier: F) -> &mut Self
    where
        F: Fn(http::Request<()>) -> O + Send + Sync + 'static,
        O: IntoFuture<Output = http::Request<()>> + 'static,
        O::IntoFuture: Send,
    {
        self.request_modifier = Some(Arc::new(move |request| {
            let request_modifier = request_modifier(request).into_future();
            Box::pin(async { request_modifier.await })
        }));
        sel
```

File: rumqttc/src/v5/mod.rs (hunk 2 above)
Comment (swanandx) [reply rumqttc/src/v5/mod.rs]: Didn't get which line exactly is this, but if removed, it won't work. I updated it as per the `Arc<_>` changes, plz have a look once and lmk if this is fixed!

File: rumqttc/src/v5/mod.rs (hunk 3 above)
Comment (swanandx) [reply rumqttc/src/v5/mod.rs]: same as about comment!

**Review by de-sh — APPROVED** [review]

**Pull request merged**